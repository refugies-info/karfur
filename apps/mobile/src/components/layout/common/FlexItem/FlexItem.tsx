import React from "react";
import { StyleProp, View, ViewStyle } from "react-native";
import styled from "styled-components/native";
import { ColumnsSpacing } from "../../Columns";

// We should use type-converting comparison (eqeq aka ==) because parseFloat will compute an invalid Float for case like: "8px" => 8
// eslint-disable-next-line eqeqeq
const isFloat = (value: any) => parseFloat(value) == value;

interface Props {
  style?: StyleProp<ViewStyle>;
  children?: any;
  flex?: string;
  RTLBehaviour?: boolean;
  marginHorizontal?: ColumnsSpacing;
  marginBottom?: "nospace" | "default" | "text";
}
// TODO: marginBotton and marginRight API are perfectible: it's already declared as an update tech task in our backlog !
const FlexItem = styled(({ style, children }: Props) => (
  <View style={style}>{children}</View>
))<{
  flex?: string;
  marginBottom?: "nospace" | "default" | "text";
  marginRight?: string;
}>`
  flex-grow: ${({ flex }) => (isFloat(flex) ? flex : 0)};
  flex-basis: ${({ flex }) => (!isFloat(flex) ? flex : 0)};
  ${({ marginBottom, theme }) =>
    marginBottom && `margin-bottom: ${theme.layout.rows[marginBottom]}`};
  ${({ marginHorizontal, RTLBehaviour, theme }) =>
    marginHorizontal &&
    (theme.i18n.isRTL && RTLBehaviour
      ? `margin-left: ${theme.layout.columns[marginHorizontal]}`
      : `margin-right: ${theme.layout.columns[marginHorizontal]}`)};
`;

export default FlexItem;
