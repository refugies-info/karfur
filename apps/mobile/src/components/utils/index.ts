export * from "./array";
export { isDarkColor } from "./isDarkColor";
