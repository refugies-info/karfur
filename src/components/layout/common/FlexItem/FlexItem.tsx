import React from "react";
import { View } from "react-native";
import styled from "styled-components/native";

// We should use type-converting comparison (eqeq aka ==) because parseFloat will compute an invalid Float for case like: "8px" => 8
// eslint-disable-next-line eqeqeq
const isFloat = (value: any) => parseFloat(value) == value;

// TODO: marginBotton and marginRight API are perfectible: it's already declared as an update tech task in our backlog !
const FlexItem = styled(({ style, children }) => (
  <View style={style}>{children}</View>
))<{ flex?: string; marginBottom?: string; marginRight?: string }>`
  flex-grow: ${({ flex }) => (isFloat(flex) ? flex : 0)};
  flex-basis: ${({ flex }) => (!isFloat(flex) ? flex : 0)};
  ${({ marginBottom, theme }) =>
    marginBottom && `margin-bottom: ${theme.layout.rows[marginBottom]}`};
  ${({ marginHorizontal, RTLBehaviour, theme }) =>
    marginHorizontal &&
    (theme.i18n.isRTL && RTLBehaviour
      ? `margin-left: ${theme.layout.columns[marginHorizontal]}`
      : `margin-right: ${theme.layout.columns[marginHorizontal]}`)};
`;

export default FlexItem;
