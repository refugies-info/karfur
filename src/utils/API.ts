import axios from "react-native-axios";
import { getEnvironment } from "../libs/getEnvironment";

const dbURL = getEnvironment().dbUrl;
const siteSecret = getEnvironment().siteSecret;

const apiCaller = axios.create({
  baseURL: dbURL,
  headers: {
    "Content-Type": "application/json",
    "site-secret": siteSecret,
  },
});

export const getLanguages = () => apiCaller.get("/langues/getLanguages");

export const getCitiesFromGoogleAPI = (text: string) =>
  apiCaller.post(
    `https://maps.googleapis.com/maps/api/place/autocomplete/json?input=${text}&language=fr&types=(cities)&components=country:fr&key=${process.env.GOOGLE_API_KEY}`
  );

export const getCityDetailsFromGoogleAPI = (placeId: string) =>
  apiCaller.post(
    `https://maps.googleapis.com/maps/api/place/details/json?place_id=${placeId}&fields=address_component&key=${process.env.GOOGLE_API_KEY}`
  );

export const getPlaceIdFromLocationFromGoogleAPI = (
  longitude: number,
  latitude: number
) =>
  apiCaller.post(
    `https://maps.googleapis.com/maps/api/geocode/json?latlng=${latitude},${longitude}&key=${process.env.GOOGLE_API_KEY}`
  );

export const getContentsForApp = ({
  locale,
  age,
  department,
  frenchLevel,
}: {
  locale: string;
  age: string | null;
  department: string | null;
  frenchLevel: string | null;
}) => {
  const route =
    `/dispositifs/getContentsForApp?locale=${locale}` +
    (age ? `&age=${age}` : "") +
    (department ? `&department=${department}` : "") +
    (frenchLevel ? `&frenchLevel=${frenchLevel}` : "");

  return apiCaller.get(route);
};

export const get_dispositif = (params = {}) =>
  apiCaller.post("/dispositifs/get_dispositif", params);
