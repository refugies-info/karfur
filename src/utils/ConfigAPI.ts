import axios from "react-native-axios";
import Config from "../libs/getEnvironment";

const dbURL = Config.dbUrl;
const siteSecret = Config.siteSecret;
export const apiCaller = axios.create({
  baseURL: dbURL,
  headers: {
    "Content-Type": "application/json",
    "site-secret": siteSecret,
  },
});