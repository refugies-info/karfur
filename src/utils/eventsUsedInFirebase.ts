export const events = {
  REACHED_END_CONTENT: "REACHED_END_CONTENT",
  LANGUAGE_VALIDATE: "LANGUAGE_VALIDATE",
  LOCATION_VALIDATE: "LOCATION_VALIDATE",
};
