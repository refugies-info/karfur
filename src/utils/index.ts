export { default as withProps } from "./withProps";
