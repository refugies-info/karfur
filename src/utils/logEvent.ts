/* eslint-disable no-console */
import * as Analytics from "expo-firebase-analytics";
import { getEnvironment } from "../libs/getEnvironment";
import { FirebaseEvent } from "./eventsUsedInFirebase";

export const logEventInFirebase = async (
  eventName: FirebaseEvent,
  data: Record<string, any>
) => {
  const { envName, debugModeFirebase } = getEnvironment();
  if (envName === "DEVELOPMENT") {
    if (debugModeFirebase === "activated") {
      Analytics.setDebugModeEnabled(true);
      console.log(
        "Environment is dev and debug mode is activated for firebase"
      );
    } else {
      console.log(
        "Environment is dev and debug mode is NOT activated for firebase"
      );
      return;
    }
  }

  if (envName && ["DEVELOPMENT", "STAGING", "PROD"].includes(envName)) {
    await Analytics.logEvent(eventName, { ...data, envName });
  }
};
