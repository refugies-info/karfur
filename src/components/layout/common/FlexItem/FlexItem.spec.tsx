import React from "react";
import FlexItem from "./FlexItem";
import { render } from "../../../utils/tests";

describe("FlexItem snapshot test suite", () => {
  it("should render without bug", () => {
    expect(
      render(<FlexItem flex="1" marginBottom="nospace" marginRight="default" />)
    ).toMatchSnapshot();
  });

  it("should accept non float value as flex prop", () => {
    expect(render(<FlexItem flex="50px" />)).toMatchSnapshot();
  });
});
