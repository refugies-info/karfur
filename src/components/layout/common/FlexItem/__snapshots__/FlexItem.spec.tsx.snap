// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlexItem snapshot test suite should accept non float value as flex prop 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 50,
        "flexGrow": 0,
      },
    ]
  }
/>
`;

exports[`FlexItem snapshot test suite should render without bug 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "marginBottom": 0,
      },
    ]
  }
/>
`;
